### ISSUE TITLE
WatchedDate and IsWatched are wrongly set to '1970-01-01 00:00:00Z' and '1' on import

### ISSUE THREAD

**Issue by krbrs** [ShokoAnime/ShokoServer#914]
## VERSION INFORMATION

Server Version: latest daily

Desktop Version: latest daily

## LOG FILE

```
12:42:39| --- ImportFolder: import || /srv/mirc/_anime/
,12:42:39| --- Processing File 1/1 --- /srv/mirc/_anime/[SubsPlease] Lupin Sansei_ Part 6 - 04 [C0DFDDC6].mkv
,12:42:39| --- Found 1 new files
,12:42:39| --- Found 1 videos
,12:42:39| --- Checking File For Hashes: /srv/mirc/_anime/[SubsPlease] Lupin Sansei_ Part 6 - 04 [C0DFDDC6].mkv
,General Queue state change: Getting file info from UDP API: [SubsPlease] Lupin Sansei_ Part 6 - 04 [C0DFDDC6].mkv
,12:42:44| --- Processing File: 58129
,12:42:44| --- Checking for AniDB_File record for: 93D3B8011CBE0A8381620B53DA47D36F --- [SubsPlease] Lupin Sansei_ Part 6 - 04 [C0DFDDC6].mkv
,12:42:44| --- ReadWatched: True File Watched: 1
,12:42:44| --- Setting watched date for User 1 and videolocal id 58129 to 01/01/1970 01:00:00
,12:42:44| --- Setting watched date for User 2 and videolocal id 58129 to 01/01/1970 01:00:00
```

## DESCRIPTION

@da3dsoul kero_chan from discord here; your said the reason might be your commit (we talked about this a while ago): https://github.com/ShokoAnime/ShokoServer/commit/de709852ae6eaa33a1b1c3b30293010c62f777b3

Please just think of this as a reminder for it :-)

When a file is added, which was never added before and neither watched (new release), the following happens in the SQLite DB:

Table AniDB_File:

WatchedDate is set to "1970-01-01 00:00:00Z"
IsWatched is set to "1"

Table VideoLocal_User:

WatchedDate is set to "1970-01-01 00:00:00Z"

This also happens when rescanning a file from Desktop which was already in the DB.

Older entries normally are 'NULL' or '0'. 

This issue causes the "Sync Plex watch status" to mark all newly added episodes as watched, even though they are not watched.

I normally use a webhook, so this doesn't affect me in general, but sometimes it does not sync the state (e.g. downloaded and offline watched media, watched via VPN and connection was lost, ...) and I manually push the state to Shoko with the "sync" button - and then it messes up my plex watched state.

(I just removed WatchedDate and IsWatched from another 400 entries in the DB manually 😅 )

Please let me know if you need further information or logs/screenshots etc.

## STEPS TO REPRODUCE

It happens when importing a new file

**Comment by da3dsoul** [comment]
Oh, yeah. My bad.

**Comment by natyusha** [comment]
This issue ended up affecting thousands of files for me so for those who are looking to fix their database the following commands seem to have sorted it out.

```SQL
UPDATE AniDB_File SET IsWatched=0, WatchedDate=NULL WHERE (WatchedDate like '1970%' OR WatchedDate like '1969%');
UPDATE VideoLocal_User SET WatchedDate=NULL WHERE (WatchedDate like '1970%' OR WatchedDate like '1969%');
UPDATE AnimeSeries_User SET UnwatchedEpisodeCount=(UnwatchedEpisodeCount+WatchedEpisodeCount), WatchedCount=0, WatchedDate=NULL WHERE (WatchedDate like '1970%' OR WatchedDate like '1969%');
UPDATE AnimeGroup_User SET UnwatchedEpisodeCount=(UnwatchedEpisodeCount+WatchedEpisodeCount), WatchedCount=0, WatchedDate=NULL WHERE (WatchedDate like '1970%' OR WatchedDate like '1969%');
UPDATE AnimeEpisode_User SET PlayedCount=0, WatchedCount=0, WatchedDate=NULL WHERE (WatchedDate like '1970%' OR WatchedDate like '1969%');
```

**Comment by revam** [comment]
The root cause of the issue was fixed by 1de2292.

**Comment by krbrs** [comment]
Be careful. This:

```
UPDATE AnimeSeries_User SET UnwatchedEpisodeCount=(UnwatchedEpisodeCount+WatchedEpisodeCount), WatchedEpisodeCount=0, WatchedCount=0, WatchedDate=NULL WHERE (WatchedDate like '1970%' OR WatchedDate like '1969%');
UPDATE AnimeGroup_User SET UnwatchedEpisodeCount=(UnwatchedEpisodeCount+WatchedEpisodeCount), WatchedEpisodeCount=0, WatchedCount=0, WatchedDate=NULL WHERE (WatchedDate like '1970%' OR WatchedDate like '1969%');
```

Caused my series to have more "UnwatchedEpisodeCount" episodes than actually present on AniDB in my test-db 😉
I just needed to set the Unwatched/Watched EpisodeCount to 0 there
